api/src/main/java/org/openmrs/module/reporting/cohort/definition/evaluator/ConditionCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,70 @@
+package org.openmrs.module.reporting.cohort.definition.evaluator;
+
+import java.util.List;
+
+import org.openmrs.Cohort;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.ConditionCohortDefinition;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Handler(supports = { ConditionCohortDefinition.class })
+public class ConditionCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+	
+	@Autowired
+	EvaluationService evaluationService;
+	
+	@Override
+	public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) {
+		
+		ConditionCohortDefinition cd = (ConditionCohortDefinition) cohortDefinition;
+		StringBuilder sql = new StringBuilder();
+		SqlQueryBuilder query = new SqlQueryBuilder();
+		
+		String sqlDefault = "SELECT patient_id FROM conditions c";
+		String sqlDate = "";
+		String sqlConditonValue = "";
+		sql.append(sqlDefault);
+		if (cd.getOnOrAfter() != null && !(cd.getOnOrBefore() != null)) {
+			sqlDate = " WHERE c.date_created >= :onOrAfter";
+		}
+		if (cd.getOnOrBefore() != null && !(cd.getOnOrAfter() != null)) {
+			sqlDate = " WHERE c.date_created <= :onOrBefore ";
+		}
+		if (cd.getOnOrBefore() != null && cd.getOnOrAfter() != null) {
+			sqlDate = " WHERE c.date_created >= :onOrAfter AND c.date_created <= :onOrBefore ";
+		}
+		if (cd.getConcept() != null) {
+			sqlConditonValue = "c.concept_id = :conceptId";
+		}
+		if (cd.getConditionNonCoded() != null) {
+			sqlConditonValue += " AND c.condition_non_coded = :conditonNonCoded ";
+		}
+		sql.append(sqlDate);
+		if (!(cd.getOnOrBefore() != null && cd.getOnOrAfter() != null) && cd.getConcept() != null) {
+			sql.append(" WHERE " + sqlConditonValue);
+		} else if ((cd.getOnOrBefore() != null || cd.getOnOrAfter() != null) && cd.getConcept() != null) {
+			sql.append(" AND " + sqlConditonValue);
+		}
+		query.append(sql.toString());
```

Comment:
I strongly recommend you look at how the HqlQueryBuilder is used in other cohort definition evaluators, and to use it here.  It greatly simplifies the writing of queries like this and handles a lot of common errors that arise.

Reply:
yes @mseaton , surely i firsttried using the HQLQueryBuilder , however i came up with several issues that i explained above and on the ticket comments